### ISSUE TITLE
Provide extra_specs property on flavors

### ISSUE THREAD

**Issue by rgbkrk** [rackspace/gophercloud#434]
As noted in https://github.com/jrperritt/rack/issues/35, extra specs is not set on the Flavors struct.

**Comment by rgbkrk** [comment]
The [definition within Rackspace's docs for what extra specs they provide](http://docs.rackspace.com/servers/api/v2/cs-gettingstarted/content/nova_list_flavors.html):

```
extra_specs
  Extra specifications

  Class
    flavor class: standard1, compute1, io1, memory1, or onmetal1

  Disk_io_index
    Relative measure of disk I/O performance from 0-99, where higher is faster

  Number_of_data_disks
    Number of data disks
```

**Comment by jrperritt** [comment]
closed by #446
